lib/stdlib/src/erl_lint.erl

```diff
@@ -1251,12 +1263,16 @@ check_imports(Forms, St0) ->
             St0;
         true ->
             Usage = St0#lint.usage,
+
             Unused = ordsets:subtract(St0#lint.imports, Usage#usage.imported),
-            Imports = [{{FA,Mod},Anno} ||
-		 {attribute,Anno,import,{Mod,Fs}} <- Forms,
-		 FA <- lists:usort(Fs)],
+            Imports = [{{FA,Mod},Anno} || {attribute,Anno,import,{Mod,Fs}} <- Forms, FA <- lists:usort(Fs)],
             Bad = [{FM,Anno} || FM <- Unused, {FM2,Anno} <- Imports, FM =:= FM2],
-            func_location_warning(unused_import, Bad, St0)
+            St1 = func_location_warning(unused_import, Bad, St0),
+
+            TUnused = ordsets:subtract(St1#lint.imp_types, Usage#usage.imported_types),
+            TImports = [{{TA,Mod},L} || {attribute,L,import_type,{Mod,Ts}} <- Forms, TA <- lists:usort(Ts)],
+            TBad = [{FM,L} || FM <- TUnused, {FM2,L} <- TImports, FM =:= FM2],
```

Comment:
One more proposal to avoid the cartesian product of two lists doing `=:=` and also trying to side-step the `lists:usort()`.

```erlang
Map = #{ {TA, Mod} => L ||  {attribute,L,import_type,{Mod,Ts}} <- Forms, TA <- Ts},
[{FM, L} || FM <- TUnused, {ok, L} <- [maps:find(FM, Map)]],
```

Reply:
The logic of linting imported types is the same as the logic of linting imported function (lines above). I tried to be close to the existing code.